dot/core/test_helpers.go

```diff
@@ -105,6 +107,17 @@ func NewTestService(t *testing.T, cfg *Config) *Service {
 		rtCfg.CodeHash, err = cfg.StorageState.LoadCodeHash(nil)
 		require.NoError(t, err)
 
+		nodeStorage := runtime.NodeStorage{}
+
+		if stateSrvc != nil {
+			nodeStorage.BaseDB = stateSrvc.Base
+		} else {
+			nodeStorage.BaseDB, err = utils.SetupDatabase(filepath.Join(testDatadirPath, "offline_storage"), false)
+			require.NoError(t, err)
+		}
+
+		rtCfg.NodeStorage = nodeStorage
```

Comment:
lgmt! just this pice of code that repeats in some places, would be nice to create a test function helper that receives a `stateSrvc` and returns a `NodeStorage`

Reply:
I created this:
```
func InitNodeStorage(baseDB runtime.BasicStorage, dataDir string) (runtime.NodeStorage, error) {
    nodeStorage := runtime.NodeStorage{}
    var err error
    if baseDB != nil {
        nodeStorage.BaseDB = baseDB
    } else {
        nodeStorage.BaseDB, err = utils.SetupDatabase(filepath.Join(dataDir, "offline_storage"), false)
    }
    return nodeStorage, err
}
```
however it seems that I could find a place to put this where it wouldn't cause a cycle import error in one of the files that used it.

Reply:
where did you place it? If you could not find a place to put there is no problem to keep the code as it is.

Reply:
Tried using it in core/test_helpers.go, then sync/test_helpers.go and babe/babe_test.go